staging/src/k8s.io/apimachinery/pkg/util/validation/validation_test.go

```diff
@@ -557,7 +557,8 @@ func TestIsFullyQualifiedDomainName(t *testing.T) {
 		"bbc.co.uk",
 		"10.0.0.1", // DNS labels can start with numbers and there is no requirement for letters.
 		"hyphens-are-good.k8s.io",
-		strings.Repeat("a", 246) + ".k8s.io",
+		strings.Repeat("a", 63) + ".k8s.io",
+		strings.Repeat("a", 63) + "." + strings.Repeat("b", 63) + "." + strings.Repeat("c", 63) + "." + strings.Repeat("d", 54) + ".k8s.io",
```

Comment:
How about a test case with a trailing '.' which I think will fail.

Reply:
Same comment with above. The trailing dot, which passes the test case, already exists since before
https://github.com/kubernetes/kubernetes/blob/413bc1a1d238efb7c4ba9e3aac2c381c93295aec/staging/src/k8s.io/apimachinery/pkg/util/validation/validation_test.go#L554